avro/src/main/java/com/fasterxml/jackson/dataformat/avro/AvroMapper.java

```diff
@@ -126,4 +146,85 @@ public AvroSchema schemaFrom(File schemaFile) throws IOException
         return new AvroSchema(new Schema.Parser().setValidate(true)
                 .parse(schemaFile));
     }
+
+    public static class AvroSchemaTypeResolverBuilder extends StdTypeResolverBuilder {
+
+        @Override
+        public TypeSerializer buildTypeSerializer(SerializationConfig config, JavaType baseType, Collection<NamedType> subtypes) {
+            // No default typing for Serialization - It's already encoded in the Schema object
+            return null;
+        }
+
+        @Override
+        public TypeDeserializer buildTypeDeserializer(DeserializationConfig config, JavaType baseType, Collection<NamedType> subtypes) {
```

Comment:
I really do not understand what is being attempted here... type id handling should start from `AnnotationIntrospector` that determines if polymorphic type handling is needed. Code here appears to assume that each and every type is polymorphic?